test/index.js

```diff
@@ -80,10 +80,35 @@ const expected = `{
 }
 `
 
-const output = prettier.format(input, {
-  filepath: 'package.json',
-  plugins: ['.'],
+;['package.json', 'foo/bar/package.json', 'foo\\bar\\package.json'].forEach(filepath => {
+  const output = prettier.format(input, {
+    filepath,
+    plugins: ['.'],
+  })
+  console.log('Testing', filepath)
+  console.assert(output === expected, 'Output does not match expected output')
+  if (output !== expected) {
+    process.exitCode = 1
+  }
+  console.log()
 })
+;['Package.json', 'package.JSON', 'package-lock.json', 'composer.json', 'package.json/composer.json'].forEach(
```

Comment:
Could you please add `undefined` or `null` to the list as well? Because we are manipulating `options.filepath` as a string, we should make sure Prettier does not throw an exception if this value is not provided (guess this can happen when the plugin is used on the front end). This would be a protection against someone removing `options.filepath && ` from the `if` you've added in `index.js`.

Reply:
If `null` is passed, prettier simply crashes. I did add a test for `undefined`.